### ISSUE TITLE
Error in In-depth FIT Tutorial of tabular 

### ISSUE THREAD

**Issue by zhanghang1989** [awslabs/autogluon#231]
Originally posted in https://github.com/Talented-zack/AutoGLuon-V.S-XGBoost/issues/1

# Forwarding the message:

Hi, all! Very excited to see that Amazon launches the Auto-GLuon algorithms. I am doing a tabular data prediction project, and I tried two different code: The first is without setting parameters: predictor = task.fit(train_data=train_data, label=label_column, output_directory=dir) This line of code runs well;

However, when I read the In-depth FIT Tutorial of tabular prediction section, I wrote my second line of code: predictor = task.fit(train_data=train_data, label=label_column, output_directory=dir, num_trials=num_trials,hyperparameter_tune=True, hyperparameters=hyperparameters) It shows the error: Auto-GLuon error:Exception in worker process: Can't pickle local object 'TaskScheduler._run_dist_job.._worker'

The parameters inside the task.fit has been set, so I doubt the error is due to the packages I missed.

#First version
#Auto-GLuon without setting parameters
label_column='default.payment.next.month'
print("Summary of class variable: \n", train_data[label_column].describe())
dir = 'agModels-predictClass' # specifies folder where to store trained models
predictor = task.fit(train_data=train_data, label=label_column, output_directory=dir)

#Second Version
#Auto-GLuon with setting parameters
label_column='default.payment.next.month'
dir = 'agModels-predictClass' # specifies folder where to store trained models
hp_tune = True # whether or not to do hyperparameter optimization

nn_options = { # specifies non-default hyperparameter values for neural network models
'num_epochs': 10, # number of training epochs (controls training time of NN models)
'learning_rate': ag.space.Real(1e-4, 1e-2, default=5e-4, log=True), # learning rate used in training (real-valued hyperparameter searched on log-scale)
'activation': ag.space.Categorical('relu', 'softrelu', 'tanh'), # activation function used in NN (categorical hyperparameter, default = first entry)
'layers': ag.space.Categorical([100],[1000],[200,100],[300,200,100]),

#Each choice for categorical hyperparameter 'layers' corresponds to list of sizes for each NN layer to use
'dropout_prob': ag.space.Real(0.0, 0.5, default=0.1), # dropout probability (real-valued hyperparameter)
}

gbm_options = { # specifies non-default hyperparameter values for lightGBM gradient boosted trees
'num_boost_round': 100, # number of boosting rounds (controls training time of GBM models)
'num_leaves': ag.space.Int(lower=26, upper=66, default=36), # number of leaves in trees (integer hyperparameter)
}

hyperparameters = {'NN': nn_options, 'GBM': gbm_options} # hyperparameters of each model type

### If one of these keys is missing from hyperparameters dict, then no models of that type are trained.

num_trials = 5 # try at most 3 different hyperparameter configurations for each type of model
search_strategy = 'skopt' # to tune hyperparameters using SKopt Bayesian optimization routine
predictor = task.fit(train_data=train_data, label=label_column, output_directory=dir, num_trials=num_trials,hyperparameter_tune=True, hyperparameters=hyperparameters

**Comment by zhanghang1989** [comment]
Hi @jwmueller, do you have any thoughts on this error?

**Comment by jwmueller** [comment]
> Hi @jwmueller, do you have any thoughts on this error?

responding in the original thread to ask for more information

**Comment by luyifanlu** [comment]
set hyperparameter_tune=False,it may be work

**Comment by Innixma** [comment]
Issue should be resolved by #920
